hub/src/server/drivers/AzDriver.ts

```diff
@@ -180,6 +183,28 @@ class AzDriver implements DriverModel, DriverModelTestMethods {
     logger.debug(`Storing ${azBlob} in ${this.bucket}, URL: ${publicURL}`)
     return publicURL
   }
+
+  async performDelete(args: PerformDeleteArgs): Promise<void> {
+    // cancel write and return 402 if path is invalid
+    if (!AzDriver.isPathValid(args.path)) {
+      throw new BadPathError('Invalid Path')
+    }
+    const azBlob = `${args.storageTopLevel}/${args.path}`
+    const blobURL = azure.BlobURL.fromContainerURL(this.container, azBlob)
+    const blockBlobURL = azure.BlockBlobURL.fromBlobURL(blobURL)
+    try {
+      await blockBlobURL.delete(azure.Aborter.none)
+    } catch (error) {
+      if (error.statusCode === 404) {
+        throw new BadPathError('File does not exist')
```

Comment:
This isn't an error due to the path being bad -- this is an error due to the file not existing.  Do we have a `DoesNotExist` exception somewhere that we can throw?

Reply:
I'll implement one